cmake/CeleritasLibrary.cmake

```
@@ -83,6 +128,37 @@ define_property(TARGET PROPERTY CELERITAS_CUDA_OBJECT_LIBRARY
   FULL_DOCS "Name of the object (without nvlink step) library corresponding to this cuda library"
 )
 
+##############################################################################
+# Separate the OPTIONS out from the sources
+#
+macro(CUDA_GET_SOURCES_AND_OPTIONS _sources _cmake_options _options)
+  set( ${_sources} )
+  set( ${_cmake_options} )
+  set( ${_options} )
+  set( _found_options FALSE )
+  foreach(arg ${ARGN})
+    if("x${arg}" STREQUAL "xOPTIONS")
```

All these need to be 
```suggestion
    if(arg STREQUAL "OPTIONS")
```